### PR TITLE
Add device type to signup args

### PR DESCRIPTION
**Pull request by patrickxb**
r? @maxtaco 
cc @cjb

### CODE REVIEW

File: go/engine/signup.go [hunk 1]
```diff
@@ -187,6 +188,9 @@ func (s *SignupEngine) registerDevice(a libkb.LoginContext, ctx *Context, device
 		Lks:        s.lks,
 		IsEldest:   true,
 	}
+	if s.arg.DeviceType == keybase1.DeviceType_MOBILE {
```
Comment (maxtaco) [inline go/engine/signup.go]: why not just always set it?

Comment (maxtaco) [inline go/engine/signup.go]: Won't there be a bug if we introduce another type? We won't think to update this line.

**Comment by patrickxb** [comment]
@maxtaco PTAL

**Comment by patrickxb** [comment]
@keybase/react-hackers I need some flow help here.  Moved DeviceType to common.

**Comment by MarcoPolo** [comment]
Fixed flow errors caused by changes to signup protocol and moving device type.

**Pull request merged**